media/media.go

```diff
@@ -143,21 +159,34 @@ func (sm svgManager) getFullPath(mediaPath string) string {
 	return filepath.Join(sm.baseDir, mediaPath)
 }
 
-func (sm svgManager) ensureDownloaded(mediaPath string) error {
+func (sm svgManager) ensureDownloaded(mediaPath string, w, h int, c color.Color) (string, error) {
+	// use the colour as a directory to separate different colours of SVG's
+	customizedMediaPath := path.Join(fmt.Sprintf("c%sw%dh%d", colorToHex6(c), w, h), mediaPath)
+
 	// check if we need to download the SVG from the media server
-	_, err := os.Stat(sm.getFullPath(mediaPath))
+	_, err := os.Stat(sm.getFullPath(customizedMediaPath))
 	if errors.Is(err, fs.ErrNotExist) {
 		svg, err := getCachedMediaResource(mediaPath)
 		if err != nil {
-			return err
+			return "", err
 		}
-		err = sm.writeFile(mediaPath, []byte(svg))
+
+		svg = customiseSVG(svg, w, h, c)
+
+		err = sm.writeFile(customizedMediaPath, []byte(svg))
 		if err != nil {
-			return err
+			return "", err
 		}
 	}
 
-	return nil
+	// return the new media path which includes the colour directory
+	return customizedMediaPath, nil
+}
+
+// customiseSVG wraps the SVG with an outer `svg` tag to ensure that it has the
+// specified width, height and fill attributes.
+func customiseSVG(svg string, w, h int, c color.Color) string {
+	return fmt.Sprintf(`<svg xmlns="http://www.w3.org/2000/svg" fill="%s" width="%d" height="%d">%s</svg>`, colorToHex6(c), w, h, svg)
```

Comment:
Does inkscape require the width & height attrs to exist? Or does it do sizing separately?

Reply:
Note: it does work if I just set `fill` on the existing SVG element, because it has the `viewBox` set, so it doesn't need height/width.

The reasons I didn't go with this approach were:
- Go doesn't have a great library for directly editing SVG's, so we need to use a generic XML library
- Go's XML processing libraries are pretty verbose and it would be relatively complex compared to the simple wrapping approach. The main benefit of not wrapping is we could save cache/disk space by not needing to set width/height.

Reply:
I have re-implemented the snake SVG colorization using an XML modifying approach